src/workerd/api/web-socket.c++

```diff
@@ -12,6 +12,30 @@
 
 namespace workerd::api {
 
+bool WebSocketProtocolError::decodeFromException(const kj::Exception& ex) {
+  kj::String magic = kj::heapString(magicFileValue);
```

Comment:
If I'm reading this correctly (which I might not be given that I've only had one cup of coffee so far this morning) ... this really shouldn't be using `kj::heapString`, especially if `magic` is only being used for comparison here. You can make this something like `auto magic = "whatever the magic string is"_kj;` to construct a `kj::StringPtr` without doing a heap allocation.

Reply:
Works for me. Done.